notes-app/note.js

```diff
@@ -0,0 +1,27 @@
+var mongoose = require('mongoose');
+
+mongoose.connect('mongodb://localhost/notesapp');
+
+var NoteSchema = mongoose.Schema({
+	title: {
+		type: String,
+		index: true
+	},
+	body: {
+		type: String
+	}
+});
+
+
+
+var Note = module.exports = mongoose.model('Note', NoteSchema);
+
+module.exports.getNoteByTitle = function(title, callback){
+	var query = {title: title};
+	Note.findOne(query, callback);
+}
+
+
+module.exports.createNote = function(newNote, callback){
```

Comment:
Can you just make sure that the notes don't have the same title? Perhaps, check for that in the database and then create the note.